cli/clone.ts

```diff
@@ -0,0 +1,130 @@
+import type {ArgumentsCamelCase} from 'yargs';
+import type {Client} from '../helpers/external-api/coda';
+import {createCodaClient} from './helpers';
+import {formatEndpoint} from './helpers';
+import fs from 'fs-extra';
+import {getApiKey} from './config_storage';
+import {handleInit} from './init';
+import {isResponseError} from '../helpers/external-api/coda';
+import {parsePackIdOrUrl} from './link';
+import path from 'path';
+import {print} from '../testing/helpers';
+import {printAndExit} from '../testing/helpers';
+import {promptForInput} from '../testing/helpers';
+import {storePackId} from './config_storage';
+
+interface CloneArgs {
+  packIdOrUrl: string;
+  codaApiEndpoint: string;
+  packVersion?: string;
+}
+
+export async function handleClone({packIdOrUrl, codaApiEndpoint}: ArgumentsCamelCase<CloneArgs>) {
+  const manifestDir = process.cwd();
+  const packId = parsePackIdOrUrl(packIdOrUrl);
+  if (!packId) {
+    return printAndExit(`Not a valid pack ID or URL: ${packIdOrUrl}`);
+  }
+
+  const formattedEndpoint = formatEndpoint(codaApiEndpoint);
+
+  const apiKey = getApiKey(codaApiEndpoint);
+  if (!apiKey) {
+    printAndExit('Missing API key. Please run `coda register <apiKey>` to register one.');
+  }
+
+  const codeAlreadyExists = fs.existsSync(path.join(process.cwd(), 'pack.ts'));
+  if (codeAlreadyExists) {
+    const shouldOverwrite = promptForInput('A pack.ts file already exists. Do you want to overwrite it? (y/N)?');
+    if (!shouldOverwrite.toLocaleLowerCase().startsWith('y')) {
+      return printAndExit('Aborting');
+    }
+  }
+
+  const client = createCodaClient(apiKey, formattedEndpoint);
+
+  let packVersion: string;
+  try {
+    const maybeVersion = await getPackLatestVersion(client, packId);
```

Comment:
Can probably just assign directly to packVersion now.

Reply:
Done